lib/ansible/modules/windows/win_group_member.ps1

```diff
@@ -0,0 +1,230 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+# WANT_JSON
+# POWERSHELL_COMMON
+
+$ErrorActionPreference = "Stop"
+
+function Test-GroupMember {
+    <#
+    .SYNOPSIS
+    Parse desired member into domain and username.
+    Also, ensure member can be resolved/exists on the target system by checking its SID.
+    .NOTES
+    Returns a hashtable of the same type as returned from Get-GroupMember.
+    Accepts username (users, groups) and domains in the following formats:
+    - username
+    - .\username
+    - SERVERNAME\username
+    - NT AUTHORITY\username
+    - DOMAIN\username
+    - username@DOMAIN
+    #>
+    param(
+        [String]$GroupMember
+    )
+
+    $parsed_member = @{
+        domain = $null
+        username = $null
+        combined = $null
+    }
+
+    # Split domain and account name into separate values
+    # '\' or '@' needs additional parsing, otherwise assume local computer
+
+    if ($GroupMember -match "\\") {
+        # DOMAIN\username
+        $split_member = $GroupMember.Split("\")
+
+        if ($split_member[0] -in @($env:COMPUTERNAME, ".")) {
+            # Local
+            $parsed_member.domain = $env:COMPUTERNAME
+        }
+        else {
+            # Domain or service (i.e. NT AUTHORITY)
+            $parsed_member.domain = $split_member[0]
+        }
+        $parsed_member.username = $split_member[1]
+    }
+    elseif ($GroupMember -match "@") {
+        # username@DOMAIN
+        $parsed_member.domain = $GroupMember.Split("@")[1]
+        $parsed_member.username = $GroupMember.Split("@")[0]
+    }
+    else {
+        # Local
+        $parsed_member.domain = $env:COMPUTERNAME
+        $parsed_member.username = $GroupMember
+    }
+
+    if ($parsed_member.domain -match "\.") {
+        # Assume FQDN was passed - change to NetBIOS/short name for later ADSI membership comparisons
+        $netbios_name = (Get-CimInstance -ClassName Win32_NTDomain -Filter "DnsForestName = '$($parsed_member.domain)'").DomainName
+
+        if (!$netbios_name) {
+            Fail-Json -obj $result -message "Could not resolve NetBIOS name for domain $($parsed_member.domain)"
+        }
+        $parsed_member.domain = $netbios_name
+    }
+
+    # Set SID check arguments, and 'combined' for later comparison and output reporting
+    if ($parsed_member.domain -eq $env:COMPUTERNAME) {
+        $sid_check_args = @($parsed_member.username)
+        $parsed_member.combined = "{0}" -f $parsed_member.username
```

Comment:
Hey @jborean93, thanks for looking into this.  I just tested this on a domain joined machine, and my testing actually shows the opposite.  It looks like when a user exists only in the domain, and NTAccount/SID translation is performed with only `username`, the domain user is returned.  However, upon the creation/existence of a local account with same username as the domain user sAMAccountName, it instead returns the local user.

Here's a test script I used:
```powershell
function Test-SidResolution {
    param($UserToTest)
    $sid_check_args = @($UserToTest)
    $user_object = New-Object -TypeName System.Security.Principal.NTAccount -ArgumentList $sid_check_args
    return $user_object.Translate([System.Security.Principal.SecurityIdentifier]).Value
}

# test user - only exists in domain initially...
$user = "testuser"
$val1 = Test-SidResolution -UserToTest $user

# create local user with same username (e.g. same as domain sAMAccountName)
$objOu = [ADSI]”WinNT://$env:COMPUTERNAME“
$objUser = $objOU.Create(“User“, $user)
$objUser.SetInfo()

# test user again after creating locally
$val2 = Test-SidResolution -UserToTest $user

"`n$val1`n" | Write-Output
"`n$val2`n" | Write-Output

S-1-5-21-<domain>-17761


S-1-5-21-<localComputer>-1008
```

With that being said, I could see a case where a false positive occurs with the current code.  e.g. A lookup is done for (an intended) local user, and only using `username` it returns the user exists, however they exist only in the domain and not locally.  I think removing the conditional argument building in 89-96, and simply instantiating NTAccount with `domain` and `username` every time (including when domain is `$env:COMPUTERNAME`) would solve this.

Windows doesn't allow nesting of local groups - and therefore, this module doesn't - so excluding that case, doing SID resolution for users (local and domain), groups (domain) and service accounts should be possible in every case with domain and username.

Let me know what you think.  If that makes sense to you, I could change this.